pkg/util/process/collector.go

```diff
@@ -0,0 +1,133 @@
+package process
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// DefaultProcMountPoint is the common mount point of the proc filesystem.
+	DefaultProcMountPoint = "/proc"
+)
+
+type processCollector struct {
+	pid            int
+	procMountPoint string
+	supported      bool
+
+	// Metrics.
+	currMaps *prometheus.Desc
+	maxMaps  *prometheus.Desc
+}
+
+// NewProcessCollector makes a new custom process collector used to collect process metrics the
+// default instrumentation doesn't support.
+func NewProcessCollector() prometheus.Collector {
+	return newProcessCollector(os.Getpid(), DefaultProcMountPoint)
+}
+
+func newProcessCollector(pid int, procMountPoint string) prometheus.Collector {
+	c := &processCollector{
+		pid:            pid,
+		procMountPoint: procMountPoint,
+		supported:      isSupported(procMountPoint),
```

Comment:
Super-Nit: If not supported, this method should return empty collector or nil. No need for having `supported` field and check it on each collection.